File: tessoku_py/B06.py
N = int(input())
A = list(map(int, input().split()))

sum = [0]
for i in range(N):
  sum.append(sum[-1] + A[i])

Q = int(input())
L = []
R = []
for q in range(Q):
  l, r = map(int, input().split())
  L.append(l)
  R.append(r)

for q in range(Q):
  attempt = R[q] - L[q] + 1
  atari = sum[R[q]] - sum[L[q]-1]
  hazure = attempt - atari
  if atari > hazure:
    print('win')
  elif atari < hazure:
    print('lose')
  else:
    print('draw')

File: tessoku_py/B06.cpp
#include <iostream>
using namespace std;

int N, A[100009], S[100009];
int Q, L[100009], R[100009];
int atari, hazure;

int main(){
  cin >> N;
  for (int i=1; i<=N; i++) cin >> A[i];
  cin >> Q;
  for (int i=1; i<=Q; i++) cin >> L[i] >> R[i];

  S[0] = 0;
  for (int i=1; i<=N; i++) S[i] = S[i-1] + A[i];

  for (int i=1; i<=Q; i++) {
    atari = S[R[i]] - S[L[i]-1];
    hazure = R[i]-L[i]+1-atari;
    if (atari>hazure) {
      cout << "win" << endl;
    } else if (atari==hazure) {
      cout << "draw" << endl;
    } else {
      cout << "lose" << endl;
    }
    
  }
  return 0;
}